src/modules/0.0.10/aws_vpc/index.ts

```diff
@@ -398,5 +428,149 @@ export const AwsVpcModule: Module2 = new Module2({
+      source: 'db',
+      cloud: new Crud2({
+        create: async (es: EndpointGateway[], ctx: Context) => {
+          const out = [];
+          const client = await ctx.getAwsClient() as AWS;
+          for (const e of es) {
+            const input: CreateVpcEndpointCommandInput = {
+              VpcEndpointType: 'Gateway',
+              ServiceName: await client.getVpcEndpointGatewayServiceName(e.service),
+              VpcId: e.vpc?.vpcId,
+            };
+            if (e.policyDocument) {
+              input.PolicyDocument = e.policyDocument;
+            }
+            if (e.routeTableIds?.length) {
+              input.RouteTableIds = e.routeTableIds;
+            } else {
+              const vpcRouteTables = await client.getVpcRouteTables(e.vpc?.vpcId ?? '');
+              input.RouteTableIds = vpcRouteTables?.map(rt => rt.RouteTableId ?? '')?.filter(id => !!id) ?? [];
+            }
+            if (e.tags && Object.keys(e.tags).length) {
+              const tags: Tag[] = Object.keys(e.tags).map((k: string) => {
+                return {
+                  Key: k, Value: e.tags![k],
+                }
+              });
+              input.TagSpecifications = [
+                {
+                  ResourceType: 'vpc-endpoint',
+                  Tags: tags,
+                },
+              ];
+            }
+            const res = await client.createVpcEndpointGateway(input);
+            const rawEndpointGateway = await client.getVpcEndpointGateway(res?.VpcEndpointId ?? '');
+            const newEndpointGateway = await AwsVpcModule.utils.endpointGatewayMapper(rawEndpointGateway, ctx);
+            newEndpointGateway.id = e.id;
+            await AwsVpcModule.mappers.endpointGateway.db.update(newEndpointGateway, ctx);
+            out.push(newEndpointGateway);
+          }
+          return out;
+        },
+        read: async (ctx: Context, id?: string) => {
+          const client = await ctx.getAwsClient() as AWS;
+          if (!!id) {
+            const rawEndpointGateway = await client.getVpcEndpointGateway(id);
+            if (!rawEndpointGateway) return;
+            return await AwsVpcModule.utils.endpointGatewayMapper(rawEndpointGateway, ctx);
+          } else {
+            const out = [];
+            for (const eg of (await client.getVpcEndpointGateways())) {
+              out.push(await AwsVpcModule.utils.endpointGatewayMapper(eg, ctx));
+            }
+            return out;
+          }
+        },
+        updateOrReplace: (a: EndpointGateway, b: EndpointGateway) => {
+          // Update vpc endpoint
+          // !Object.is(a.policyDocument, b.policyDocument)
+          // !Object.is(a.routeTableIds, b.routeTableIds)
+          // Update tags
+          // !AwsVpcModule.utils.eqTags(a.tags, b.tags)
+          // Restore
+          // !Object.is(a.state, b.state)
+          // Replace
+          // !Object.is(a.vpc?.vpcId, b.vpc?.vpcId)
+          // !Object.is(a.service, b.service)
```

Comment:
Delete this?

Reply:
Oh that was my guide to see what need to be done in each case. I'll delete it